oshc/main/tests.py

```diff
@@ -1,6 +1,63 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
-from django.test import TestCase
+from django.contrib.auth.models import User
+from django.test import SimpleTestCase, TestCase
+from django.urls import reverse
 
-# Create your tests here.
+
+class EmailLoginBackendTests(TestCase):
+
+    def setUp(self):
+        self.credentials = {
+            'username': 'testuser',
+            'email': 'testuser@email.com',
+            'password': 'secret'}
+        User.objects.create_user(**self.credentials)
+
+    def test_valid_username_login(self):
+        response = self.client.login(username=self.credentials['username'],
+                                     password=self.credentials['password'])
+        self.assertTrue(response)
+
+    def test_invalid_username_login(self):
+        response = self.client.login(username='invalid_username',
+                                     password=self.credentials['password'])
+        self.assertFalse(response)
+
+    def test_valid_email_login(self):
+        response = self.client.login(username=self.credentials['email'],
+                                     password=self.credentials['password'])
+        self.assertTrue(response)
+
+    def test_invalid_email_login(self):
+        response = self.client.login(username='invalid_email',
+                                     password=self.credentials['password'])
+        self.assertFalse(response)
+
+    def test_invalid_password_login(self):
+        response = self.client.login(username=self.credentials['email'],
+                                     password='incorrect_password')
+        self.assertFalse(response)
+
+
+class HomeViewTests(SimpleTestCase):
+
+    def test_get_request(self):
+        response = self.client.get(reverse("home"))
+        self.assertEqual(response.status_code, 200)
```

Comment:
@tapasweni-pathak This test will raise an error if `home` responds with anything other than `200`. I don't think we need to test for not success statuses.